python/cudf/cudf/_libxx/rolling.pyx

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from __future__ import print_function
+import cudf
+import pandas as pd
+import numba
+
+from cudf._libxx.lib cimport *
+from cudf._libxx.column cimport *
+
+from cudf._libxx.aggregation cimport *
+
+from cudf._libxx.includes.rolling cimport (
+    rolling_window as cpp_rolling_window
+)
+
+
+def rolling(Column source_column, window, min_periods, center, op):
+    """
+    Rolling on input executing operation within the given window for each row
+
+    Parameters
+    ----------
+    source_column : input column on which rolling operation is executed
+    window : Size of the moving window, can be integer or numba DeviceArray
+    min_periods : Minimum number of observations in window required to have
+                  a value (otherwise result is null)
+    center : Set the labels at the center of the window
+    op : operation to be executed, as of now it supports MIN, MAX, COUNT, SUM,
+         MEAN and UDF
+
+    Returns
+    -------
+    A Column with rolling calculations
+    """
+
+    cdef size_type c_min_periods = min_periods
+    cdef size_type c_window = 0
+    cdef size_type c_forward_window = 0
+    cdef unique_ptr[column] c_result
+
+    if isinstance(window, numba.cuda.cudadrv.devicearray.DeviceNDArray):
+        if center:
+            # TODO: we can support this even though Pandas currently does not
+            raise NotImplementedError(
+                "center is not implemented for offset-based windows"
+            )
+        column_window = cudf.Series(window)._column
+        f_column_window = cudf.Series([0] * column_window.size,
+                                      dtype=column_window.dtype)._column
+
+        pre_column_window = Column(
+            column_window.data, column_window.size,
+            column_window.dtype)
```

Comment:
This looks like just shallow copying `column_window`, is that the case?

Reply:
Yes, it is just shallow copying. I will try to use as_column instead.

Reply:
Do we need a shallow copy? If so you can just do `col.copy(deep=False)`

Reply:
Think, I was getting some error as it couldn't determine type of object while calling cpp function, but will give it a try.